app/src/main/java/com/libs/globaldata/GlobalData.java

```diff
@@ -49,4 +55,19 @@ public static void clearModel(String modelName, Context context) {
             }
         }
     }
+
+    /**
+     * Get All user settings defined in SharedPreferences.
+     *
+     * @param context - app/activity context
+     * @return userSettings - preferences with all user settings
+     */
+    public static SharedPreferences getUserSettings(Context context){
+        if(userSettings == null){
+            synchronized (ModelObject.class) {
+                userSettings = context.getSharedPreferences(context.getResources().getString(R.string.user_settings_name), Context.MODE_PRIVATE);

```

Comment:
I think there is another "if(userSettings == null){" required after the synchronized block. Thread waiting for sychronized section to unblock might proceed through the first if and then overrite the settings.